src/main/java/com/github/manolo8/darkbot/core/api/GameAPIImpl.java

```diff
@@ -464,6 +466,14 @@ public void selectEntity(Entity entity) {
             if (entity instanceof Lockable) {
                 //assuming that selectEntity selects only ships & is supported by every API
                 //actually this should be called on every entity with LockType trait
+
+                if (mapManager.isTarget(entity)) return;
+                if (!(entity instanceof Ship) || entity instanceof MapNpc) {
+                    if (!mapManager.setTarget(entity.address)) {
+                        return;
+                    }
+                } /*else mapManager.setTarget(0);*/
```

Comment:
```suggestion
                if ((!(entity instanceof Ship) || entity instanceof MapNpc) && !mapManager.setTarget(entity.address)) 
                    return;
```
isn't it functionally identical?